docs/src/modules/components/AppSearch.js

```diff
@@ -124,13 +142,27 @@ export default function AppSearch() {
 
   React.useEffect(() => {
     const handleKeyDown = (nativeEvent) => {
-      if (
-        ['/', 's'].indexOf(nativeEvent.key) !== -1 &&
-        document.activeElement.nodeName === 'BODY' &&
-        document.activeElement !== inputRef.current
-      ) {
-        nativeEvent.preventDefault();
-        inputRef.current.focus();
+      if (nativeEvent.key === 'Escape' && document.activeElement === inputRef.current) {
+        inputRef.current.blur();
+        return;
+      }
+
+      // Match shortcuts of GitHub and DocSearch
+      const matchKey =
+        ((nativeEvent.ctrlKey || nativeEvent.metaKey) && nativeEvent.key === 'k') ||
+        nativeEvent.key === '/' ||
+        nativeEvent.key === 's';
```

Comment:
Why do we need 3 keyboard shortcuts?

Especially `s` is problematic. The WCAG guideline to avoid single keys exists for a reason. For example, try our   Select components type-ahead feature. On https://deploy-preview-23959--material-ui.netlify.app/components/selects/#basic-select focus the select, open it, and press `s`. The searchbox will now take focus and our trapfocus will return it to the container i.e. it will focus an element that is not matching what the user typed. On `next` nothing happens as expected (https://next--material-ui.netlify.app/components/selects/#basic-select).

Reply:
Oh nice, we need to improve the logic that filters active elements.

Reply:
> WCAG guideline to avoid single keys exists for a reason.

The reason seems to be about speech users, by a large margin. They dedicate a lot of the content to it. I couldn't find our problem covered. 

Regarding the solution, I could find 3 options:

1. We ignore it.
1. We only handle the keyboard shortcut when body and main are focused.
1. We add `.defaultMuiPrevented` to signal that the event was already handled.

I'm happy with any of these 3. with a preference for 2. Thoughts?

Reply:
> We ignore it.

So we're re-opening the discussion about WCAG. We settled on AA by default but it seems like you're going to reject every single rule as I'm opening them. What is your principled approach to WCAG to which I can hold you accountable? You're always very concerned with ROI but if we don't make decision we're wasting a lot of time in discussions.

Reply:
> We settled on AA by default

@eps1lon I would add a distinction: we settled on it for the components. Did we discuss the documentation yet? I can't remember it. I think that, so far, we fixed issues on the documentation that were preventing correctly audit the accessibility of the components, but without having a strong focus on the documentation itself. I think that it deserves a different discussion because the incentives are not obviously identical. For the component, the motivation is that developers need it for the end-users or from legal perspectives.

I think that it's similar to the discussion about screenreaders support. We didn't talk about it initially, but later on, we said that we would support the most recent version of the most common screen readers, as well as document it in https://material-ui.com/getting-started/supported-platforms/. At least, from what I recall.

> You're always very concerned with ROI but if we don't make a decision we're wasting a lot of time in discussions.

You are right, one of my concerns with WCAG is the return on investment. I think that there is a tradeoff to find between doing enough to be leading the field and not doing too much, sinking resources. My second concern is when it requires harming the UX of the majority of the people. So far, it felt that working on WCAG improved the UX most of the time. The cases where it goes against the UX of unimpaired users are not frequent. Sometimes it's by a small margin, like here. I think that we should be cautious when it's significant.